megamek/src/megamek/client/ui/swing/MovementDisplay.java

```diff
@@ -3396,6 +3399,61 @@ private Coords getUnloadPosition(Entity unloaded) {
         }
         return choice;
     }
+    
+    /**
+     * Uses player input to find a legal hex where an EjectedCrew unit can be placed
+     * @param abandoned - The vessel we're escaping from
+     * @param crew - The EjectedCrew unit we're placing
+     * @return
+     */
+    private Coords getEjectPosition(Entity abandoned) {
+        // we need to allow the user to select a hex for offloading the unit's crew
+        Coords pos = abandoned.getPosition();
+        //Create a bogus crew entity to use for legal hex calculation
+        Entity crew = new EjectedCrew();
+        crew.setId(clientgui.getClient().getGame().getNextEntityId());
+        crew.setGame(clientgui.getClient().getGame());
+        int elev = clientgui.getClient().getGame().getBoard().getHex(pos).getLevel() + abandoned.getElevation();
+        ArrayList<Coords> ring = Compute.coordsAtRange(pos, 1);
+        if (abandoned instanceof Dropship) {
+            ring = Compute.coordsAtRange(pos, 2);
+        }
+        // ok now we need to go through the ring and identify available
+        // Positions
+        ring = Compute.getAcceptableUnloadPositions(ring, crew, clientgui
+                .getClient().getGame(), elev);
+        if (ring.size() < 1) {
+            String title = Messages
+                    .getString("MovementDisplay.NoPlaceToEject.title"); //$NON-NLS-1$
+            String body = Messages
+                    .getString("MovementDisplay.NoPlaceToEject.message"); //$NON-NLS-1$
+            clientgui.doAlertDialog(title, body);
+            return null;
+        }
+        String[] choices = new String[ring.size()];
+        int i = 0;
+        for (Coords c : ring) {
+            choices[i++] = c.toString();
+        }
+        String selected = (String) JOptionPane.showInputDialog(clientgui,
+                                                               Messages.getString(
+                                                                       "MovementDisplay.ChooseEjectHex" + ".message", new Object[]{//$NON-NLS-1$
```

Comment:
Gotta ask, why are we adding the two strings together here?

Reply:
Because that's how the UnloadUnit Dialog I borrowed this from did it.  It works as one string, so fixed.